lib/diagnostic.js

```diff
@@ -6,26 +6,40 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const normalizedWords = function getNormalizedWords(stringInput) {
+  let words = stringInput.split(/\s+/);
+  for(let i=0; i<words.length; i++){
+    words[i] = words[i].replace(/\W+/, '').toUpperCase();
+  }
+  return words;
 };
 
+
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWords = function normalizedWords(stringInput) {
+  let uniqueWords = {};
+  let normalizedWordsList = normalizedWords(stringInput);
+  for(let i = 0; i < normalizedWordsList.length; i++){
+    uniqueWords[normalizedWordsList[i]] = 1;
+  }
+  return Object.keys(uniqueWords);
 };
 
```

Comment:
OH Man! THIS IS WHY! 
On line 22 you are actually calling the function you're in (making it recursive). This ends up blowing up the stack because it's just continually calling itself. Recursive functions are tricky, and not something in the scope of this course to my knowledge, but if you want to investigate them feel free to check out the interwebs. In the meantime, just leave the name off on the right side of the equals, `const uniqueWords = function(stringInput) {` or make sure you're calling it the same thing: `const uniqueWords = function uniqueWords(stringInput) {`